### ISSUE TITLE
Bug: The plugin's configurations use relative paths to include eachother

### ISSUE THREAD

**Issue by maritaria** [typescript-eslint/typescript-eslint#5007]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Issue Description

Extending the eslint config from `plugin:@typescript-eslint/strict` creates configuration load error, rendering eslint unable to run at all.

The problem here is that the configs extend eachother using`./configs/base` (or some other config instead of 'base').
But this only works for the top-level configuration being evaluated by eslint.
With the new 'strict' config, a new level is added (strict -> recommended -> base).
The extension on top of base fails because the relative path is now not starting from `dist/index.js` but `dist/configs/recommended.js`).

Fix is to replace `./configs/?` with `plugin:@typescript-eslint/?`, so it always resolves through the exports offered by the plugin.

### Reproduction Repository Link

https://github.com/maritaria/eslint-typescript-strict-repro

### Repro Steps

1) clone the repo
2) `yarn install`
3) `yarn test`, expected output: Failure to load configuration
4) `yarn patch`
5) `yarn test`, expected output: the rendered configuration of `./.eslintrc.js`

The problem is that for eslint@^6 and eslint@^7 the resolution of nested configurations fails for the new `strict` configuration the `@typescript-eslint/eslint-plugin` package offers.

This happens because the configurations include eachother as `./configs/${name}`, which works for the top-level inclusion (at `@typescript-eslint/eslint-plugin/dist/index.js`) because the module that exposes the configurations is in the correct path relatively.
But because `strict` includes `recommended` the `base` and `eslint-recommended` configurations are included as a result and those do not exist relatively to the `recommended`.

```shell
yarn remove eslint && yarn add eslint@^6.0.0 && yarn test
yarn remove eslint && yarn add eslint@^7.0.0 && yarn test
yarn remove eslint && yarn add eslint@^8.0.0 && yarn test
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.24.0` |
| `@typescript-eslint/parser`            | `5.24.0` |
| `TypeScript`                           | `4.6.4` |
| `ESLint`                               | `6.8.0`, `7.32.0`, `8.15.0` |
| `node`                                 | `v14.19.2`, `v16.14.2` |
